test/e2e/delete_test.go

```diff
@@ -35,25 +35,36 @@ func testWindowsNodeDeletion(t *testing.T) {
 	}
 
 	require.NotNil(t, windowsMachineSetWithLabel, "could not find MachineSet with Windows label")
-	// Set the expected number of Windows nodes to 0
-	gc.numberOfNodes = 0
 
 	// Scale the Windows MachineSet to 0
-	windowsMachineSetWithLabel.Spec.Replicas = &gc.numberOfNodes
+	expectedNodeCount := int32(0)
+	windowsMachineSetWithLabel.Spec.Replicas = &expectedNodeCount
 	_, err = testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Update(context.TODO(),
 		windowsMachineSetWithLabel, meta.UpdateOptions{})
 	require.NoError(t, err, "error updating Windows MachineSet")
 
 	// we are waiting 10 minutes for all windows machines to get deleted.
-	err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, true, false)
+	err = testCtx.waitForWindowsNodes(expectedNodeCount, true, false, false)
 	require.NoError(t, err, "Windows node deletion failed")
 
+	t.Run("BYOH node removal", func(t *testing.T) {
+		assert.NoError(t, testCtx.deleteWindowsInstanceConfigMap(),
+			"error removing windows-instances ConfigMap")
```

Comment:
nit: "error removing Windows instances from ConfigMap"

Reply:
The "windows-instances" ConfigMap is a specific configmap :smiley: